packages/platform-android/native_modules.gradle

```diff
@@ -150,37 +153,42 @@ class ReactNativeModules {
     if (this.reactNativeModules != null) return this.reactNativeModules
     ArrayList<HashMap<String, String>> reactNativeModules = new ArrayList<HashMap<String, String>>()
 
-    def cmdProcess
     def npx = Os.isFamily(Os.FAMILY_WINDOWS) ? "npx.cmd" : "npx"
-    def command = "${npx} --quiet react-native config"
+    def command = "${npx} --quiet --no-install react-native config"
     def reactNativeConfigOutput = ""
 
     try {
-      cmdProcess = Runtime.getRuntime().exec(command)
+      // When running Gradle from CWD that's outside of JS project, it's necessary to pass a CWD that's close to JS app,// so that npx can resolve to correct local CLI, instead of possibly installed global, corrupting config output
+      def cmdProcess = Runtime.getRuntime().exec(command, null, new File(this.jsAppDir))
```

Comment:
What about PATH variable?

Reply:
> If envp is null, the subprocess inherits the environment settings of the current process.

https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#exec-java.lang.String:A-java.lang.String:A-java.io.File-

Reply:
You are correct. But parent java process (eclipse, gradle, AndroidStudio) do not always have same PATH like you have in shell.

https://stackoverflow.com/questions/30118905/read-path-variable-using-java-in-eclipse-on-mac-os-x-differs-from-shell/36282645

For instance my Eclipse have only PATH=/usr/bin:/bin:/usr/sbin:/sbin and can't find npx.
But in shell PATH=/Users/ify/.jenv/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/opt/X11/bin:/Users/ify/Library/Android/sdk/tools:/Users/ify/Library/Android/sdk/platform-tools

Workaround is to launch npx via bash under OSX. This way same PATH like in shell will be used.
```
Runtime.getRuntime().exec("bash npx", null, new File("/"));
```

Reply:
If you could come up with a way to run it like this for Windows as well, that would be great!

Reply:
I have no Window box to test, but I expect something like this will work
```
Runtime.getRuntime().exec("cmd /C npx", null, new File("/"));
```
https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/cmd

Reply:
Now that I think about it, `bash` environment and `PATH` it holds may differ from a default shell (like `zsh` on latest macOS or `fish`). Doesn't Java has a method to execute the command in a default shell? That would make it easier

Reply:
echo $SHELL

Reply:
I didn't find a reliable way to get the `$SHELL` from Java's Runtime exec, so I'll pass on that. This is also not the scope of this PR.

The issue with not inheriting PATH seems related to the environment, and there are people discussing this in SO:
- https://stackoverflow.com/questions/40168112/path-is-different-when-i-run-bash-from-java-processbuilder
- https://stackoverflow.com/questions/603785/environment-variables-in-mac-os-x

For example, I don't have troubles running this script through Android Studio on macOS Catalina (10.15). Would appreciate any additional help in this field though :)